app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollView.kt

```diff
@@ -0,0 +1,269 @@
+package nerd.tuxmobil.fahrplan.congress.schedule
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.GestureDetector
+import android.view.GestureDetector.SimpleOnGestureListener
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import android.widget.HorizontalScrollView
+import android.widget.LinearLayout
+import androidx.annotation.IntRange
+import androidx.annotation.VisibleForTesting
+import androidx.core.view.children
+import androidx.core.view.get
+import info.metadude.android.eventfahrplan.commons.logging.Logging
+import nerd.tuxmobil.fahrplan.congress.R
+import kotlin.math.abs
+import kotlin.math.ceil
+import kotlin.math.floor
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+
+class HorizontalSnapScrollView(
+    context: Context,
+    attrs: AttributeSet
+) : HorizontalScrollView(context, attrs) {
+
+    companion object {
+        private const val LOG_TAG = "HorizontalScrollView"
+        private const val NOT_INITIALIZED = Int.MIN_VALUE
+        private const val SWIPE_MIN_DISTANCE = 5
+        private const val SWIPE_THRESHOLD_VELOCITY = 2800
+        private const val MIN_PORTRAIT_SCROLLING_PERCENTAGE = 0.25
+
+        /**
+         * Calculates the number of columns to display at a time based on the physical dimensions and
+         * the screen density of the device and the column count of the schedule. Further limiting
+         * factors are a maximum column count to be displayed and a minimum column width.
+         */
+        @VisibleForTesting
+        fun calculateDisplayColumnCount(
+            availablePixels: Int,
+            columnsCount: Int,
+            maxColumnCountForLayout: Int,
+            densityScaleFactor: Float,
+            minColumnWidthDip: Int
+        ): Int {
+            val columnCountLimit = min(columnsCount, maxColumnCountForLayout)
+            if (columnCountLimit == 1) {
+                return 1
+            }
+            val availableDips = availablePixels.toFloat() / densityScaleFactor
+            val minWidthColumnCount = floor((availableDips / minColumnWidthDip).toDouble()).toInt()
+            val columnCount = min(minWidthColumnCount, columnCountLimit)
+            return max(1, columnCount)
+        }
+    }
+
+    private val gestureDetector: GestureDetector
+    private val logging: Logging = Logging.get()
+    private var xStart = 0
+    private var roomsCount = NOT_INITIALIZED
+    private var roomNames: HorizontalScrollView? = null
```

Comment:
Did you try if `roomNames` could be non-nullable? It is set in `FahrplanFragment` via `snapScroller.setChildScroller(roomScroller)` where `roomScroller` is guaranteed to be non-null. Not sure if anyone else is accessing `roomNames` before.